libraries/ESP_NOW/src/ESP32_NOW_Serial.cpp

```diff
@@ -0,0 +1,272 @@
+#include "ESP32_NOW_Serial.h"
+#include <string.h>
+#include "esp_now.h"
+#include "esp_system.h"
+#include "esp32-hal.h"
+#include "esp_mac.h"
+
+/*
+ *
+ *    Serial Port Implementation Class
+ *
+*/
+
+ESP_NOW_Serial_Class::ESP_NOW_Serial_Class(const uint8_t *mac_addr, uint8_t channel, wifi_interface_t iface, const uint8_t *lmk)
+: ESP_NOW_Peer(mac_addr, channel, iface, lmk){
+    tx_ring_buf = NULL;
+    rx_queue = NULL;
+    tx_sem = NULL;
+    queued_size = 0;
+    queued_buff = NULL;
+    resend_count = 0;
+}
+
+ESP_NOW_Serial_Class::~ESP_NOW_Serial_Class(){
+    end();
+}
+
+size_t ESP_NOW_Serial_Class::setTxBufferSize(size_t tx_queue_len){
+    if(tx_ring_buf){
+        if(!tx_queue_len){
+            vRingbufferDelete(tx_ring_buf);
+            tx_ring_buf = NULL;
+        }
+        return 0;
+    }
+    tx_ring_buf = xRingbufferCreate(tx_queue_len, RINGBUF_TYPE_BYTEBUF);
+    if(!tx_ring_buf){
+        return 0;
+    }
+    return tx_queue_len;
+}
+
+size_t ESP_NOW_Serial_Class::setRxBufferSize(size_t rx_queue_len){
+    if(rx_queue){
+        if(!rx_queue_len){
+            vQueueDelete(rx_queue);
+            rx_queue = NULL;
+        }
+        return 0;
+    }
```

Comment:
The code fails, deleting the TX Ringbuffer if used this way:
``` cpp
    NowSerial.setRxBufferSize(512));
    NowSerial.begin(115200);
// or like this:
    NowSerial.begin(115200);
    NowSerial.setRxBufferSize(512));
// or just calling it twice, like this:
    NowSerial.setRxBufferSize(512));
    NowSerial.setRxBufferSize(512));
```
```suggestion
    if(rx_queue){
        vQueueDelete(rx_queue);
        rx_queue = NULL;
    }
    if(!rx_queue_len){
        return 0;
    }
```